test/membrane_tcp/sink/pipeline_test.exs

```diff
@@ -0,0 +1,148 @@
+defmodule Membrane.TCP.SinkPipelineTest do
+  use ExUnit.Case, async: false
+
+  import Membrane.ChildrenSpec
+
+  alias Membrane.TCP.{Sink, Socket}
+  alias Membrane.Testing.{Pipeline, Source}
+
+  @localhost {127, 0, 0, 1}
+  @server_port_no 5052
+
+  @payload_frames 100
+  @data Enum.map(1..@payload_frames, &"(#{&1})") ++ ["."]
+
+  defp run_pipeline(pipeline, socket) do
+    received_data = receive_data(socket)
+    assert received_data == Enum.join(@data)
+    Pipeline.terminate(pipeline)
+    Socket.close(socket)
+  end
+
+  defp receive_data(socket, received_data \\ "", terminator \\ ".") do
+    {:ok, payload} = Socket.recv(socket)
+
+    if String.ends_with?(payload, terminator) do
+      received_data <> payload
+    else
+      receive_data(socket, received_data <> payload)
```

Comment:
shouldn't you also pass the `terminator` here? :p

Reply:
no, because the payload already ends with terminator